lib/grape/request.rb

```diff
@@ -47,7 +47,11 @@ def build_headers
     end
 
     def transform_header(header)
-      -header[5..].split('_').each(&:capitalize!).join('-')
+      if Gem::Version.new(Rack.release) < Gem::Version.new('3')
```

Comment:
This executes in a tight loop, which is bad for performance. You want this check outside. We also have it in a few places. It should look like this:

```ruby
if Grape::Rack.rack3?
  def transform_header(header)
      -header[5..].split('_').map(&:capitalize).join('-')
  end
else
  def transform_header(header)
      -header[5..].tr('_', '-').downcase
  end
end
```

Reply:
Added `.rack3?` method and done this change, although I've added it to `Grape` rather than `Grape::Rack` otherwise there was a load of Rack errors raised because we would need to change all the Rack references to `::Rack`.

Can do this though if really needs to be `Grape::Rack`